src/middleware/apiKeyChecker.ts

```diff
@@ -0,0 +1,23 @@
+import http from 'http';
+import { ApiKeyError } from '../errors';
+import type {
+  NextFunction,
+  Request,
+  Response,
+} from 'express';
+
+export const checkApiKey = (req: Request, res: Response, next: NextFunction): void => {
```

Comment:
Could you add some integration tests for this endpoint in order to get 100% diff coverage?  You can check out the other integration tests for some good patterns,